src/main/java/org/opentestsystem/ap/ims/service/SavedSearchService.java

```diff
@@ -0,0 +1,128 @@
+package org.opentestsystem.ap.ims.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.opentestsystem.ap.common.util.SecurityUtil;
+import org.opentestsystem.ap.ims.entity.SavedSearch;
+import org.opentestsystem.ap.ims.entity.SavedSearch.SavedSearchIdentity;
+import org.opentestsystem.ap.ims.entity.SharedSavedSearch;
+import org.opentestsystem.ap.ims.model.SavedSearchModel;
+import org.opentestsystem.ap.ims.repository.SavedSearchRepository;
+import org.opentestsystem.ap.ims.repository.SharedSavedSearchRepository;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
```

Comment:
The Service annotation makes the calls transactional.  I put READONLY on the finders.